### PR TITLE
Enable nightly builds

### PR DESCRIPTION
**Pull request by mtojek**
## What does this PR do?

This PR enables nightly builds, so we can receive daily feedback on test execution.

### CODE REVIEW

**Comment by exekias** [comment]
Thanks for taking this! I'm wondering, how will errors be notified?

**Comment by mtojek** [comment]
Ideally we should open an issue for this, but I'm not sure where to put the automation for this, whether there is a generic mechanism or scripts. Any suggestions, @elastic/observablt-robots ?

Anyway, we can introduce it in follow-ups.

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -30,6 +30,7 @@ pipeline {
   }
   triggers {
     issueCommentTrigger('(?i).*(?:jenkins\\W+)?run\\W+(?:the\\W+)?tests(?:\\W+please)?.*')
+    cron 'H 02 * * *'
```
Comment (kuisathaverat) [inline .ci/Jenkinsfile:33]: better to put a range of hours, the CI will decide the less overload time.
```suggestion
    cron 'H H(2-5) * * *'
```

**Review by kuisathaverat — APPROVED** [review]

File: .ci/Jenkinsfile [hunk 2]
```diff
@@ -30,6 +30,7 @@ pipeline {
   }
   triggers {
     issueCommentTrigger('(?i).*(?:jenkins\\W+)?run\\W+(?:the\\W+)?tests(?:\\W+please)?.*')
```
Comment (kuisathaverat) [inline .ci/Jenkinsfile:32]: BTW we can add the `/test` comment as trigger.

```suggestion
    issueCommentTrigger('(?i)(.*(?:jenkins\\W+)?run\\W+(?:the\\W+)?tests(?:\\W+please)?.*|/test)')
```

**Comment by kuisathaverat** [comment]
> Thanks for taking this! I'm wondering, how will errors be notified?

at the moment this job only reports to GitHub and to the CI, if you need other kinds of notification (email or slack), please request it with an issue.

**Review by exekias — APPROVED** [review]

**Review by cachedout — APPROVED** [review]

**Comment by mtojek** [comment]
Thanks for review, @cachedout @kuisathaverat and @exekias.

**Pull request merged**